Loglama

### Sistemdeki tüm hatalar loglanacaktır.

 **Hata Logları**
` [ERROR] - [SourceErrorFunc] : "Error Message" -> Excepiton`

**Bilgilendirme Logları**
`[INFO] - [InfoMessage]`

şeklinde oluşturulacaktır.

Loglar `C:\NutrixaApp` içinde `yyyy-MM-dd.txt` olarak günlük loglanacaktır. Her gün içi ayrı bir `.txt `açılacaktır.

--- a/APP.API/Controllers/AccountsController.cs
+++ b/APP.API/Controllers/AccountsController.cs
@@ -7,6 +7,8 @@ using System;
 using System.Threading.Tasks;
 using ORM.Models.Models;
 using ORM.Models.Models.CustomModels;
+using ORM.Business.Classes;
+using System.Text.Json;
 
 namespace APP.API.Controllers
 {
@@ -33,6 +35,7 @@ namespace APP.API.Controllers
             }
             catch (Exception ex)
             {
+                NutrixaLogger.LogError($"[AccountsController-GetById] Kullanıcı ID'si: {id} ile kullanıcı bilgisi alınırken bir hata oluştu.", ex);
                 return Problem(detail: "Genel Hata: " + ex.Message, statusCode: 500);
             }
         }
@@ -46,6 +49,7 @@ namespace APP.API.Controllers
             }
             catch (Exception ex)
             {
+                NutrixaLogger.LogError("[AccountsController-GetAll] Tüm kullanıcılar alınırken bir hata oluştu.", ex);
                 return Problem(detail: "Genel Hata: " + ex.Message, statusCode: 500);
             }
         }
@@ -67,6 +71,9 @@ namespace APP.API.Controllers
             }
             catch (Exception ex)
             {
+                NutrixaLogger.LogError($"[AccountsController-Update] Kullanıcı  bilgisi güncellenirken bir hata oluştu.", ex);
+                string updatedItemJson=JsonHelper.SerializeObject(updateRequest.UpdatedItem);
+                NutrixaLogger.LogInfo($"[AccountsController-Update] Güncellenen kullanıcı bilgisi: {updatedItemJson}");
                 return Problem(detail: "Genel Hata: " + ex.Message, statusCode: 500);
             }
         }
@@ -85,6 +92,9 @@ namespace APP.API.Controllers
             }
             catch (Exception ex)
             {
+                NutrixaLogger.LogError($"[AccountsController-Add] Yeni kullanıcı eklenirken bir hata oluştu.", ex);
+                string userJson = JsonHelper.SerializeObject(user);
+                NutrixaLogger.LogInfo($"[AccountsController-Add] Eklenmeye çalışılan kullanıcı bilgisi: {userJson}");
                 return Problem(detail: "Genel Hata: " + ex.Message, statusCode: 500);
             }
         }

--- a/APP.API/Controllers/AuthenticateController.cs
+++ b/APP.API/Controllers/AuthenticateController.cs
@@ -5,6 +5,7 @@ using System.Threading.Tasks;
 using System;
 using ORM.Services.IServices;
 using ORM.Services.Models;
+using ORM.Business.Classes;
 
 namespace APP.API.Controllers
 {
@@ -19,7 +20,7 @@ namespace APP.API.Controllers
             _userService = userService;
         }
         [HttpPost]
-        public async Task<IActionResult> GetToken([FromBody] APP.API.Models.AuthenticateRequest model)
+        public async Task<IActionResult> GetToken([FromBody] AuthenticateRequest model)
         {
             try
             {
@@ -33,6 +34,9 @@ namespace APP.API.Controllers
             }
             catch (Exception ex)
             {
+                NutrixaLogger.LogError("[GetToken] Token alınırken bir hata oluştu.", ex);
+                string json = JsonHelper.SerializeObject(model);
+                NutrixaLogger.LogInfo($"[GetToken] Hata oluşan model verisi: {json}");
                 return Problem(detail: "Genel Hata: " + ex.Message, statusCode: 500);
             }
         }

--- a/APP.API/Controllers/CalculatesController.cs
+++ b/APP.API/Controllers/CalculatesController.cs
@@ -1,6 +1,7 @@
 ﻿using APP.API.Auth;
 using Microsoft.AspNetCore.Http;
 using Microsoft.AspNetCore.Mvc;
+using ORM.Business.Classes;
 using ORM.Models.Models;
 using ORM.Services.IServices;
 using ORM.Services.Services;
@@ -30,6 +31,9 @@ namespace APP.API.Controllers
             }
             catch (Exception ex)
             {
+                NutrixaLogger.LogError("[GetTargetDays] Hedef gün sayısı hesaplanırken bir hata oluştu.", ex);
+                string userJson = JsonHelper.SerializeObject(user);
+                NutrixaLogger.LogInfo($"[GetTargetDays] Hata oluşan kullanıcı verisi: {userJson}");
                 return Problem(detail: "Genel Hata: " + ex.Message, statusCode: 500);
             }
         }

--- a/APP.API/Controllers/UsersController.cs
+++ b/APP.API/Controllers/UsersController.cs
@@ -9,6 +9,7 @@ using ORM.Services.Models;
 using APP.API.Auth;
 using ORM.Models.Models;
 using ORM.Services.Services;
+using ORM.Business.Classes;
 
 namespace APP.API.Controllers
 {
@@ -26,14 +27,24 @@ namespace APP.API.Controllers
         }
 
         [HttpPost("authenticate")]
-        public async Task<IActionResult> Authenticate([FromBody] APP.API.Models.AuthenticateRequest model)
+        public async Task<IActionResult> Authenticate([FromBody] AuthenticateRequest model)
         {
-            var result = await _userService.AuthenticateByEmail(model.Email, model.Password, model.IsPasswordMd5);
+            try
+            {
+                var result = await _userService.AuthenticateByEmail(model.Email, model.Password, model.IsPasswordMd5);
 
-            if (result == null || string.IsNullOrEmpty(result.Token))
-                return BadRequest(new { message = "Kullanıcı adı veya parola yanlış." });
+                if (result == null || string.IsNullOrEmpty(result.Token))
+                    return BadRequest(new { message = "Kullanıcı adı veya parola yanlış." });
 
-            return Ok(new { token = result.Token, is_registration = result.IsRegistration });
+                return Ok(new { token = result.Token, is_registration = result.IsRegistration });
+            }
+            catch (Exception ex)
+            {
+                NutrixaLogger.LogError("[Authenticate] Kimlik doğrulaması yapılırken bir hata oluştu.", ex);
+                string json = JsonHelper.SerializeObject(model);
+                NutrixaLogger.LogInfo($"[Authenticate] Hata oluşan model verisi: {json}");
+                return Problem(detail: "Genel Hata: " + ex.Message, statusCode: 500);
+            }
         }
 
         [HttpPost("create")]
@@ -85,6 +96,9 @@ namespace APP.API.Controllers
             }
             catch (Exception ex)
             {
+                NutrixaLogger.LogError("[UsersController-Create] Kullanıcı oluşturulurken bir hata oluştu.", ex);
+                string userJson = JsonHelper.SerializeObject(model);
+                NutrixaLogger.LogInfo($"[UsersController-Create] Eklenmek istenen kullanıcı verisi: {userJson}");
                 return Problem(detail: "Genel Hata: " + ex.Message, statusCode: 500);
             }
         }

--- a/ORM.Business/Classes/JsonHelper.cs
+++ b/ORM.Business/Classes/JsonHelper.cs
@@ -1,0 +1,37 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace ORM.Business.Classes
+{
+    public class JsonHelper
+    {
+        public static string SerializeObject(object obj)
+        {
+            try
+            {
+                return System.Text.Json.JsonSerializer.Serialize(obj);
+            }
+            catch (Exception ex)
+            {
+                NutrixaLogger.LogError("[JsonHelper-SerializeObject] Nesne serileştirilirken bir hata oluştu.", ex);
+                return string.Empty;
+            }
+        }
+
+        public static T DeserializeObject<T>(string json)
+        {
+            try
+            {
+                return System.Text.Json.JsonSerializer.Deserialize<T>(json);
+            }
+            catch (Exception ex)
+            {
+                NutrixaLogger.LogError("[JsonHelper-DeserializeObject] JSON verisi nesneye dönüştürülürken bir hata oluştu.", ex);
+                return default(T);
+            }
+        }
+    }
+}

--- a/ORM.Business/Classes/NutrixaLogger.cs
+++ b/ORM.Business/Classes/NutrixaLogger.cs
@@ -44,7 +44,7 @@ namespace ORM.Business.Classes
             roller.ActivateOptions();
 
             hierarchy.Root.AddAppender(roller);
-            hierarchy.Root.Level = Level.Error;
+            hierarchy.Root.Level = Level.Info;
             hierarchy.Configured = true;
         }
 

--- a/ORM.Services/IServices/IUserService.cs
+++ b/ORM.Services/IServices/IUserService.cs
@@ -16,7 +16,6 @@ namespace ORM.Services.IServices
         Task<bool> Delete(object id);
         Task<bool> DeleteUser(users userObj);
         Task<users> GetUser(object id);
-        Task<users?> AddAndUpdateUser(object userObj);
         Task<bool> ExistsByEmail(string email);
     }
 }

--- a/ORM.Services/Services/CalculateService.cs
+++ b/ORM.Services/Services/CalculateService.cs
@@ -45,6 +45,8 @@ namespace ORM.Services.Services
             catch (Exception ex)
             {
                 NutrixaLogger.LogError("[CalculateDailyProtein] Günlük protein ihtiyacı hesaplanırken bir  hata oluştu.", ex);
+                string json=JsonHelper.SerializeObject(user);
+                NutrixaLogger.LogInfo($"[CalculateDailyProtein] Hata oluşan kullanıcı verisi: {json}");
                 return 0.0m;
             }
         }
@@ -67,12 +69,13 @@ namespace ORM.Services.Services
                         bmr += 5;
                     }
                 }
-                NutrixaLogger.LogInfo($"[CalculateBMR] BMR hesaplandı BMR : {bmr}");
+                NutrixaLogger.LogInfo($"[CalculateBMR] BMR hesaplandı BMR : {bmr} kcal");
             }
             catch (Exception ex)
             {
                 NutrixaLogger.LogError("[CalculateBMR] BMR hesaplanırken bir  hata oluştu.", ex);
-               
+               string json = JsonHelper.SerializeObject(user);
+                NutrixaLogger.LogInfo($"[CalculateBMR] Hata oluşan kullanıcı verisi: {json}");
             }
             return bmr;
         }
@@ -109,6 +112,8 @@ namespace ORM.Services.Services
             catch (Exception ex)
             {
                 NutrixaLogger.LogError("[CalculateTDEE] Günlük kalori ihtiyacı hesaplanırken bir  hata oluştu.", ex);
+                string json = JsonHelper.SerializeObject(user);
+                NutrixaLogger.LogInfo($"[CalculateTDEE] Hata oluşan kullanıcı verisi: {json}");
                 return 0.0m;
             }
         }
@@ -126,6 +131,8 @@ namespace ORM.Services.Services
             catch (Exception ex)
             {
                 NutrixaLogger.LogError("[CalculateBMI] BMI hesaplanırken bir  hata oluştu.", ex);
+                string json = JsonHelper.SerializeObject(user);
+                NutrixaLogger.LogInfo($"[CalculateBMI] Hata oluşan kullanıcı verisi: {json}");
                 return 0.0m;
             }
         }
@@ -148,7 +155,8 @@ namespace ORM.Services.Services
             catch (Exception ex)
             {
                 NutrixaLogger.LogError("[CalculateTargetDays] Hedefe kalan gün sayısı hesaplanırken bir  hata oluştu.", ex);
-
+                string json = JsonHelper.SerializeObject(user);
+                NutrixaLogger.LogInfo($"[CalculateTargetDays] Hata oluşan kullanıcı verisi: {json}");
                 return 0;
             }
         }
@@ -175,6 +183,8 @@ namespace ORM.Services.Services
             catch (Exception ex)
             {
                 NutrixaLogger.LogError("[CalculateAge] Yaş hesaplanırken bir  hata oluştu.", ex);
+                string json = JsonHelper.SerializeObject(user);
+                NutrixaLogger.LogInfo($"[CalculateAge] Hata oluşan kullanıcı verisi: {json}");
                 return age;
             }
         }

--- a/ORM.Services/Services/NutritionService.cs
+++ b/ORM.Services/Services/NutritionService.cs
@@ -28,6 +28,8 @@ namespace ORM.Services.Services
             catch (Exception ex)
             {
                 NutrixaLogger.LogError("[AddFood] Food eklenirken bir  hata oluştu.", ex);
+                string json = JsonHelper.SerializeObject(food);
+                NutrixaLogger.LogInfo($"[AddFood] Hata oluşan food verisi: {json}");
             }
             return result;
         }
@@ -51,7 +53,8 @@ namespace ORM.Services.Services
             catch (Exception ex)
             {
 
-                NutrixaLogger.LogError("[DeleteFood] Food silinirken bir  hata oluştu.", ex);
+                NutrixaLogger.LogError($"[DeleteFood] Food silinirken bir  hata oluştu. Food ID: {food_id}", ex);
+
             }
             return result;
         }
@@ -84,7 +87,7 @@ namespace ORM.Services.Services
             }
             catch (Exception ex)
             {
-                NutrixaLogger.LogError("[GetFoodsByCategory] Food listesi getirilirken bir  hata oluştu.", ex);
+                NutrixaLogger.LogError($"[GetFoodsByCategory] Food listesi getirilirken bir  hata oluştu. Category ID: {category_id}", ex);
 
             }
             return foodList;
@@ -122,6 +125,8 @@ namespace ORM.Services.Services
             {
 
                 NutrixaLogger.LogError("[UpdateFood] Food güncellenirken bir  hata oluştu.", ex);
+                string json = JsonHelper.SerializeObject(food);
+                NutrixaLogger.LogInfo($"[UpdateFood] Hata oluşan food verisi: {json}");
 
             }
             return result;

--- a/ORM.Services/Services/UserService.cs
+++ b/ORM.Services/Services/UserService.cs
@@ -11,6 +11,7 @@ using System.Security.Claims;
 using System.Text;
 using ORM.Models.Models;
 using ORM.Business.Classes;
+using Newtonsoft.Json.Linq;
 
 namespace ORM.Services.Services
 {
@@ -25,171 +26,286 @@ namespace ORM.Services.Services
 
         public async Task<users?> AddUser(users user)
         {
-            if (user == null || string.IsNullOrWhiteSpace(user.email))
+            try
+            {
+                if (user == null || string.IsNullOrWhiteSpace(user.email))
+                    return null;
+
+                // Check by email or username
+                if (!string.IsNullOrWhiteSpace(user.email) && await ExistsByEmail(user.email))
+                    return null;
+
+
+                user.id = Guid.NewGuid();
+                user.password = ORM.Shared.Base.GenerateDoubleMD5Password(user.password);
+
+                var res = new GenericBusiness<users>().Add(user);
+                return res ? user : null;
+            }
+            catch (Exception ex)
+            {
+                NutrixaLogger.LogError("[AddUser] Kullanýcý eklenirken bir  hata oluþtu.", ex);
+                string json = JsonHelper.SerializeObject(user);
+                NutrixaLogger.LogInfo($"[AddUser] Hata oluþan kullanýcý verisi: {json}");
                 return null;
+            }
 
-            // Check by email or username
-            if (!string.IsNullOrWhiteSpace(user.email) && await ExistsByEmail(user.email))
-                return null;
-
-
-            user.id = Guid.NewGuid();
-            user.password = ORM.Shared.Base.GenerateDoubleMD5Password(user.password);
-
-            var res = new GenericBusiness<users>().Add(user);
-            return res ? user : null;
         }
 
 
         public async Task<bool> ExistsByEmail(string email)
         {
-            if (string.IsNullOrWhiteSpace(email)) return false;
-            string where = $" AND {nameof(users.email)} = '{email}'";
-            var existing = new GenericBusiness<users>().GetAllByCustomQuery(where).FirstOrDefault();
-            return existing != null;
+            try
+            {
+                if (string.IsNullOrWhiteSpace(email)) return false;
+                string where = $" AND {nameof(users.email)} = '{email}'";
+                var existing = new GenericBusiness<users>().GetAllByCustomQuery(where).FirstOrDefault();
+                return existing != null;
+            }
+            catch (Exception ex)
+            {
+
+                NutrixaLogger.LogError($"[ExistsByEmail]  E-posta adresine göre kullanýcý aranýrken bir  hata oluþtu. E-posta Adresi: {email} ", ex);
+                return false;
+            }
+
         }
 
 
         public async Task<users?> GetById(object id)
         {
-            return new GenericBusiness<users>().GetById(id);
+            try
+            {
+                return new GenericBusiness<users>().GetById(id);
+
+            }
+            catch (Exception ex)
+            {
+                NutrixaLogger.LogError($"[UserService-GetById] Kullanýcý getirirlirken bir  hata oluþtu.Kullanýcý ID: {id}", ex);
+                return new users();
+            }
         }
 
         public async Task<List<users>> GetAll()
         {
-            return new GenericBusiness<users>().GetAll().ToList();
+            try
+            {
+                return new GenericBusiness<users>().GetAll().ToList();
+
+            }
+            catch (Exception ex)
+            {
+                NutrixaLogger.LogError("[UserService-GetAll] Kullanýcý Listesi getirirlirken bir  hata oluþtu.", ex);
+                return new List<users>();
+            }
         }
 
         public async Task<bool> Update(users user)
         {
-            return new GenericBusiness<users>().Update(user);
+            try
+            {
+                return new GenericBusiness<users>().Update(user);
+            }
+            catch (Exception ex)
+            {
+
+                NutrixaLogger.LogError("[UserService-Update] Kullanýcý güncellenirken bir  hata oluþtu.", ex);
+                string json = JsonHelper.SerializeObject(user);
+                NutrixaLogger.LogInfo($"[UserService-Update] Hata oluþan kullanýcý verisi: {json}");
+                return false;
+            }
         }
 
         public async Task<bool> Delete(object id)
         {
-            return new GenericBusiness<users>().Delete(id);
+            try
+            {
+                return new GenericBusiness<users>().Delete(id);
+
+            }
+            catch (Exception ex)
+            {
+                NutrixaLogger.LogError($"[UserService-Delete] Kullanýcý silinirken bir  hata oluþtu. Kullanýcý ID: {id}", ex);
+                return false;
+            }
         }
 
         public async Task<bool> DeleteUser(users userObj)
         {
-            userObj.is_active = false;
-            userObj.is_deleted = true;
-            userObj.modified_date = DateTime.Now;
-            return new GenericBusiness<users>().Update(userObj);
+            try
+            {
+                userObj.is_active = false;
+                userObj.is_deleted = true;
+                userObj.modified_date = DateTime.Now;
+                return new GenericBusiness<users>().Update(userObj);
+            }
+            catch (Exception ex)
+            {
+                NutrixaLogger.LogError("[DeleteUser] Kullanýcý aktiflik durumu deðiþtirilirken bir  hata oluþtu.", ex);
+                string json = JsonHelper.SerializeObject(userObj);
+                NutrixaLogger.LogInfo($"[DeleteUser] Hata oluþan kullanýcý verisi: {json}");
+                return false;
+            }
+
         }
 
         public async Task<users> GetUser(object id)
         {
-            return new GenericBusiness<users>().GetById(id);
+            try
+            {
+                return new GenericBusiness<users>().GetById(id);
+            }
+            catch (Exception ex)
+            {
+                NutrixaLogger.LogError($"[GetUser] Kullanýcý getirirlirken bir  hata oluþtu.Kullanýcý ID:{id}", ex);
+                return new users();
+            }
         }
 
         public async Task<users?> AddAndUpdateUser(users userObj)
         {
-            bool isSuccess = false;
-            if (userObj.id != Guid.Empty)
+            try
             {
-                var obj = new GenericBusiness<users>().GetById(userObj.id);
-                if (obj != null)
+                bool isSuccess = false;
+                if (userObj.id != Guid.Empty)
                 {
-                    isSuccess = new GenericBusiness<users>().Update(userObj);
-                }
-            }
-            else
-            {
-                string where = $" AND {nameof(users.email)} = '{userObj.email}'";
-                var user = new GenericBusiness<users>().GetAllByCustomQuery(where).FirstOrDefault();
-                if (user == null)
-                {
-                    userObj.id = Guid.NewGuid();
-                    userObj.password = ORM.Shared.Base.GenerateDoubleMD5Password(userObj.password);
-                    isSuccess = new GenericBusiness<users>().Add(userObj);
+                    var obj = new GenericBusiness<users>().GetById(userObj.id);
+                    if (obj != null)
+                    {
+                        isSuccess = new GenericBusiness<users>().Update(userObj);
+                    }
                 }
                 else
                 {
-                    return null;
+                    string where = $" AND {nameof(users.email)} = '{userObj.email}'";
+                    var user = new GenericBusiness<users>().GetAllByCustomQuery(where).FirstOrDefault();
+                    if (user == null)
+                    {
+                        userObj.id = Guid.NewGuid();
+                        userObj.password = ORM.Shared.Base.GenerateDoubleMD5Password(userObj.password);
+                        isSuccess = new GenericBusiness<users>().Add(userObj);
+                    }
+                    else
+                    {
+                        return null;
+                    }
                 }
+
+                return isSuccess ? userObj : null;
+            }
+            catch (Exception ex)
+            {
+
+                NutrixaLogger.LogError("[AddAndUpdateUser] Kullanýcý eklenirken/güncellenirken bir  hata oluþtu.", ex);
+                string json = JsonHelper.SerializeObject(userObj);
+                NutrixaLogger.LogInfo($"[AddAndUpdateUser] Hata oluþan kullanýcý verisi: {json}");
+                return new users();
             }
 
-            return isSuccess ? userObj : null;
         }
 
         public async Task<AuthenticationResult?> AuthenticateByEmail(string email, string password, bool isPasswordMd5)
         {
-            if (string.IsNullOrWhiteSpace(email) || string.IsNullOrWhiteSpace(password))
-                return null;
-
-            string identifier = email.Trim();
-
-            string md5Password = "";
-            if (!isPasswordMd5)
+            try
             {
-                md5Password = ORM.Shared.Base.GenerateDoubleMD5Password(password);
+                if (string.IsNullOrWhiteSpace(email) || string.IsNullOrWhiteSpace(password))
+                    return null;
+
+                string identifier = email.Trim();
+
+                string md5Password = "";
+                if (!isPasswordMd5)
+                {
+                    md5Password = ORM.Shared.Base.GenerateDoubleMD5Password(password);
+                }
+                else
+                {
+                    md5Password = password;
+                }
+
+                // Allow login by either email or user_name
+                string where = $" AND ( {nameof(users.email)} = '{identifier}' OR {nameof(users.email)} = '{identifier}' ) AND {nameof(users.password)} = '{md5Password}' AND {nameof(users.is_active)} = true AND {nameof(users.is_deleted)} = false";
+                var user = new GenericBusiness<users>().GetAllByCustomQuery(where).FirstOrDefault();
+
+                if (user == null) return null;
+
+                // generate token and include is_registration
+                var token = await isAuthenticate(user);
+
+                return new AuthenticationResult { Token = token, IsRegistration = true };
             }
-            else
+            catch (Exception ex)
             {
-                md5Password = password;
+                NutrixaLogger.LogError($"[AuthenticateByEmail] Email e göre kimlik doðrulama oluþturulurken bir  hata oluþtu. E-posta adresi: {email}, Þifre: {password}, MD5'li mi: {isPasswordMd5}", ex);
+                return new AuthenticationResult { Token = "", IsRegistration = false };
             }
 
-            // Allow login by either email or user_name
-            string where = $" AND ( {nameof(users.email)} = '{identifier}' OR {nameof(users.email)} = '{identifier}' ) AND {nameof(users.password)} = '{md5Password}' AND {nameof(users.is_active)} = true AND {nameof(users.is_deleted)} = false";
-            var user = new GenericBusiness<users>().GetAllByCustomQuery(where).FirstOrDefault();
-
-            if (user == null) return null;
-
-            // generate token and include is_registration
-            var token = await isAuthenticate(user);
-
-            return new AuthenticationResult { Token = token, IsRegistration = true };
-        }
-
-        public async Task<users?> AddAndUpdateUser(object userObj)
-        {
-            return await Task.FromResult<users?>(null);
         }
 
         private async Task<string> isAuthenticate(users user)
         {
-            string w = $" AND {nameof(authtoken.user_id)}='{user.id}' AND {nameof(authtoken.expire_date)}>'{DateTime.Now}'";
-            authtoken _token = new GenericBusiness<authtoken>().GetAllByCustomQuery(w).FirstOrDefault();
-            string _jwtToken = "";
-            if (_token == null)
+            try
             {
-                authtoken _authtoken = new authtoken
+                string w = $" AND {nameof(authtoken.user_id)}='{user.id}' AND {nameof(authtoken.expire_date)}>'{DateTime.Now}'";
+                authtoken _token = new GenericBusiness<authtoken>().GetAllByCustomQuery(w).FirstOrDefault();
+                string _jwtToken = "";
+                if (_token == null)
                 {
-                    id = Guid.NewGuid(),
-                    token = await generateJwtToken(user),
-                    user_id = user.id,
-                    created_date = DateTime.Now,
-                    expire_date = DateTime.Now.AddDays(1)
-                };
-                bool res = new GenericBusiness<authtoken>().Add(_authtoken);
-                _jwtToken = _authtoken.token;
+                    authtoken _authtoken = new authtoken
+                    {
+                        id = Guid.NewGuid(),
+                        token = await generateJwtToken(user),
+                        user_id = user.id,
+                        created_date = DateTime.Now,
+                        expire_date = DateTime.Now.AddDays(1)
+                    };
+                    bool res = new GenericBusiness<authtoken>().Add(_authtoken);
+                    _jwtToken = _authtoken.token;
+                }
+                else
+                {
+                    _token.expire_date = _token.expire_date.AddHours(3);
+                    bool res = new GenericBusiness<authtoken>().Update(_token);
+                    _jwtToken = _token.token;
+                }
+                return _jwtToken;
             }
-            else
+            catch (Exception ex)
             {
-                _token.expire_date = _token.expire_date.AddHours(3);
-                bool res = new GenericBusiness<authtoken>().Update(_token);
-                _jwtToken = _token.token;
+                NutrixaLogger.LogError("[isAuthenticate] Kimlik doðrulama sýrasýnda bir  hata oluþtu.", ex);
+                string json = JsonHelper.SerializeObject(user);
+                NutrixaLogger.LogInfo($"[isAuthenticate] Hata oluþan kullanýcý verisi: {json}");
+                return string.Empty;
             }
-            return _jwtToken;
+
         }
 
         private async Task<string> generateJwtToken(users user)
         {
-            var tokenHandler = new JwtSecurityTokenHandler();
-            if (string.IsNullOrEmpty(_secret) || Encoding.UTF8.GetBytes(_secret).Length < 32)
+            try
             {
-                throw new InvalidOperationException("JWT secret is not configured or too short. Set AppSettings:Secret with at least 32 bytes length (e.g. 32+ characters).");
+                var tokenHandler = new JwtSecurityTokenHandler();
+                if (string.IsNullOrEmpty(_secret) || Encoding.UTF8.GetBytes(_secret).Length < 32)
+                {
+                    throw new InvalidOperationException("JWT secret is not configured or too short. Set AppSettings:Secret with at least 32 bytes length (e.g. 32+ characters).");
+                }
+                var key = Encoding.UTF8.GetBytes(_secret);
+                var tokenDescriptor = new SecurityTokenDescriptor
+                {
+                    Subject = new ClaimsIdentity(new[] { new Claim("id", user.id.ToString()) }),
+                    Expires = DateTime.UtcNow.AddDays(7),
+                    SigningCredentials = new SigningCredentials(new SymmetricSecurityKey(key), SecurityAlgorithms.HmacSha256Signature)
+                };
+                var token = tokenHandler.CreateToken(tokenDescriptor);
+                return tokenHandler.WriteToken(token);
             }
-            var key = Encoding.UTF8.GetBytes(_secret);
-            var tokenDescriptor = new SecurityTokenDescriptor
+            catch (Exception ex)
             {
-                Subject = new ClaimsIdentity(new[] { new Claim("id", user.id.ToString()) }),
-                Expires = DateTime.UtcNow.AddDays(7),
-                SigningCredentials = new SigningCredentials(new SymmetricSecurityKey(key), SecurityAlgorithms.HmacSha256Signature)
-            };
-            var token = tokenHandler.CreateToken(tokenDescriptor);
-            return tokenHandler.WriteToken(token);
+                NutrixaLogger.LogError("[generateJwtToken] Token oluþturulurken bir  hata oluþtu.", ex);
+                string json = JsonHelper.SerializeObject(user);
+                NutrixaLogger.LogInfo($"[generateJwtToken] Hata oluþan kullanýcý verisi: {json}");
+                return string.Empty;
+            }
         }
     }
 }

